api/src/middleware/authorization-middleware.ts

```diff
@@ -0,0 +1,80 @@
+import { type NextFunction, type Response } from "express"
+import { type Request as JwtRequest } from "express-jwt"
+import { isNil } from "lodash"
+
+import logger from "@/utils/logger"
+import { User } from "@/models"
+
+import auth0Integration, { Auth0PayloadError } from "@/integrations/auth0-integration"
+
+import sequelize from "@/models"
+
+export type AuthorizationRequest = JwtRequest & {
+  user?: User
+}
+
+async function findUserFromAuth0Subject(auth0Subject: string): Promise<User | null> {
+  return User.findOne({ where: { sub: auth0Subject } })
+}
+
+export async function ensureUserFromAuth0Token(token: string): Promise<User> {
+  const { auth0Subject, email, firstName, lastName } = await auth0Integration.getUserInfo(token)
+  const user = await findUserFromAuth0Subject(auth0Subject)
```

Comment:
I'd move this function's code inline as the function is only doing a Sequelize query and nothing else. If it were doing several things, breaking it out would make sense.